src/test/java/teammates/test/cases/BaseTestCase.java

```diff
@@ -114,6 +116,25 @@ protected static void ignorePossibleException() {
         assertTrue(true);
     }
     
+    protected static String getPopulatedErrorMessage(String messageTemplate, String userInput,
+                                                     String fieldName, String errorReason) {
+        return getPopulatedErrorMessage(messageTemplate, userInput, fieldName, errorReason, 0);
+    }
+
+    protected static String getPopulatedErrorMessage(String messageTemplate, String userInput,
+                                                     String fieldName, String errorReason, int maxLength) {
+        try {
+            Method getPopulatedErrorMessage =
+                    FieldValidator.class.getDeclaredMethod("getPopulatedErrorMessage", String.class,
+                            String.class, String.class, String.class, int.class);
+            getPopulatedErrorMessage.setAccessible(true);
+            return (String) getPopulatedErrorMessage.invoke(null, messageTemplate, userInput, fieldName,
+                                                            errorReason, maxLength);
```

Comment:
Anyway, I understand your reasoning for not wanting to invoke the private method. There is some validity to that. If you wish so, you can change this method to copy the definition of `getPopulatedErrorMessage` in `FieldValidator`. My concern when you did that in `FieldValidatorTest` was that it caused strange dependency to `FieldValidatorTest`, but if the method is in `BaseTestCase` like right now, that is an acceptable dependency.